sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -291,6 +287,20 @@ private[hive] class SparkExecuteStatementOperation(
       sqlContext.sparkContext.cancelJobGroup(statementId)
     }
   }
+
+  private def withSchedulerPool[T](body: => T): T = {
+    val pool = sessionToActivePool.get(parentSession.getSessionHandle)
+    if (pool != null) {
```

Comment:
Although this looks like it undoes the previous fix, you're saying it will always be null when this method starts right? It seems easier to unconditionally set it, unless there's some overhead I'm missing

Reply:
Because only the user submitted sql has ```set spark.sql.thriftserver.scheduler.pool=xxx```, ```sessionToActivePool``` will put xxx value, will not be the default scheduler, so the default scheduler pool name is null

Unconditional setting it is worried about ambiguity, because only after the put value, there will be a need to take this value, although there is no overhead

https://github.com/apache/spark/blob/43bf4ae6417fcb15d0fbc7880f14f307c164d464/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala#L236-L239